int32 posX
int32 posY
